`hathora build` fails with import errors

Steps to reproduce:

```
npm install --global hathora@0.5.19
git clone https://github.com/hathora/among-us-tutorial
cd among-us-tutorial
hathora dev
# Ctrl-C
hathora build
```

```
Error: Build failed with 5 errors:
server/.hathora/protocol.ts:2:31: ERROR: Could not resolve "bin-serde"
server/.hathora/store.ts:3:23: ERROR: Could not resolve "chance"
server/.hathora/store.ts:4:31: ERROR: Could not resolve "bin-serde"
server/.hathora/store.ts:5:7: ERROR: Could not resolve "dotenv/config"
server/.hathora/wrapper.ts:1:21: ERROR: Could not resolve "on-change"
```

<details>

```
Project root: /tmp/among-us-tutorial
✘ [ERROR] Could not resolve "chance"

    server/.hathora/store.ts:3:23:
      3 │ import { Chance } from "chance";
        ╵                        ~~~~~~~~

  You can mark the path "chance" as external to exclude it from the bundle, which will remove this
  error.

✘ [ERROR] Could not resolve "bin-serde"

    server/.hathora/store.ts:4:31:
      4 │ import { Writer, Reader } from "bin-serde";
        ╵                                ~~~~~~~~~~~

  You can mark the path "bin-serde" as external to exclude it from the bundle, which will remove
  this error.

✘ [ERROR] Could not resolve "dotenv/config"

    server/.hathora/store.ts:5:7:
      5 │ import "dotenv/config";
        ╵        ~~~~~~~~~~~~~~~

  You can mark the path "dotenv/config" as external to exclude it from the bundle, which will remove
  this error.

✘ [ERROR] Could not resolve "bin-serde"

    server/.hathora/protocol.ts:2:31:
      2 │ import { Reader, Writer } from "bin-serde";
        ╵                                ~~~~~~~~~~~

  You can mark the path "bin-serde" as external to exclude it from the bundle, which will remove
  this error.

✘ [ERROR] Could not resolve "on-change"

    server/.hathora/wrapper.ts:1:21:
      1 │ import onChange from "on-change";
        ╵                      ~~~~~~~~~~~

  You can mark the path "on-change" as external to exclude it from the bundle, which will remove
  this error.

vite v2.8.4 building for production...
vite v2.8.4 building for production...
[!] Error: unfinished hook action(s) on exit:
(vite:load-fallback) load "/tmp/among-us-tutorial/client/prototype-ui/index.html"
(vite:load-fallback) load "/tmp/among-us-tutorial/client/web/index.html"

/home/kaspar/.nvm/versions/node/v16.13.2/lib/node_modules/hathora/node_modules/esbuild/lib/main.js:1602
  let error = new Error(`${text}${summary}`);
              ^

Error: Build failed with 5 errors:
server/.hathora/protocol.ts:2:31: ERROR: Could not resolve "bin-serde"
server/.hathora/store.ts:3:23: ERROR: Could not resolve "chance"
server/.hathora/store.ts:4:31: ERROR: Could not resolve "bin-serde"
server/.hathora/store.ts:5:7: ERROR: Could not resolve "dotenv/config"
server/.hathora/wrapper.ts:1:21: ERROR: Could not resolve "on-change"
    at failureErrorWithLog (/home/kaspar/.nvm/versions/node/v16.13.2/lib/node_modules/hathora/node_modules/esbuild/lib/main.js:1602:15)
    at /home/kaspar/.nvm/versions/node/v16.13.2/lib/node_modules/hathora/node_modules/esbuild/lib/main.js:1248:28
    at runOnEndCallbacks (/home/kaspar/.nvm/versions/node/v16.13.2/lib/node_modules/hathora/node_modules/esbuild/lib/main.js:1161:65)
    at buildResponseToResult (/home/kaspar/.nvm/versions/node/v16.13.2/lib/node_modules/hathora/node_modules/esbuild/lib/main.js:1246:7)
    at /home/kaspar/.nvm/versions/node/v16.13.2/lib/node_modules/hathora/node_modules/esbuild/lib/main.js:1355:14
    at /home/kaspar/.nvm/versions/node/v16.13.2/lib/node_modules/hathora/node_modules/esbuild/lib/main.js:665:9
    at handleIncomingPacket (/home/kaspar/.nvm/versions/node/v16.13.2/lib/node_modules/hathora/node_modules/esbuild/lib/main.js:762:9)
    at Socket.readFromStdout (/home/kaspar/.nvm/versions/node/v16.13.2/lib/node_modules/hathora/node_modules/esbuild/lib/main.js:632:7)
    at Socket.emit (node:events:390:28)
    at addChunk (node:internal/streams/readable:315:12) {
  errors: [
    {
      detail: undefined,
      location: {
        column: 31,
        file: 'server/.hathora/protocol.ts',
        length: 11,
        line: 2,
        lineText: 'import { Reader, Writer } from "bin-serde";',
        namespace: '',
        suggestion: ''
      },
      notes: [
        {
          location: null,
          text: 'You can mark the path "bin-serde" as external to exclude it from the bundle, which will remove this error.'
        }
      ],
      pluginName: '',
      text: 'Could not resolve "bin-serde"'
    },
    {
      detail: undefined,
      location: {
        column: 23,
        file: 'server/.hathora/store.ts',
        length: 8,
        line: 3,
        lineText: 'import { Chance } from "chance";',
        namespace: '',
        suggestion: ''
      },
      notes: [
        {
          location: null,
          text: 'You can mark the path "chance" as external to exclude it from the bundle, which will remove this error.'
        }
      ],
      pluginName: '',
      text: 'Could not resolve "chance"'
    },
    {
      detail: undefined,
      location: {
        column: 31,
        file: 'server/.hathora/store.ts',
        length: 11,
        line: 4,
        lineText: 'import { Writer, Reader } from "bin-serde";',
        namespace: '',
        suggestion: ''
      },
      notes: [
        {
          location: null,
          text: 'You can mark the path "bin-serde" as external to exclude it from the bundle, which will remove this error.'
        }
      ],
      pluginName: '',
      text: 'Could not resolve "bin-serde"'
    },
    {
      detail: undefined,
      location: {
        column: 7,
        file: 'server/.hathora/store.ts',
        length: 15,
        line: 5,
        lineText: 'import "dotenv/config";',
        namespace: '',
        suggestion: ''
      },
      notes: [
        {
          location: null,
          text: 'You can mark the path "dotenv/config" as external to exclude it from the bundle, which will remove this error.'
        }
      ],
      pluginName: '',
      text: 'Could not resolve "dotenv/config"'
    },
    {
      detail: undefined,
      location: {
        column: 21,
        file: 'server/.hathora/wrapper.ts',
        length: 11,
        line: 1,
        lineText: 'import onChange from "on-change";',
        namespace: '',
        suggestion: ''
      },
      notes: [
        {
          location: null,
          text: 'You can mark the path "on-change" as external to exclude it from the bundle, which will remove this error.'
        }
      ],
      pluginName: '',
      text: 'Could not resolve "on-change"'
    }
  ],
  warnings: []
}
```


</details>

--- a/cli.ts
+++ b/cli.ts
@@ -85,6 +85,31 @@ async function startFrontends() {
   }
 }
 
+function build() {
+  process.env.VITE_APP_ID = appId;
+  for (const dir of readdirSync(clientDir)) {
+    if (existsSync(join(clientDir, dir, "index.html"))) {
+      buildClient({
+        root: join(clientDir, dir),
+        build: { outDir: join(rootDir, "dist", "client", dir), target: ["esnext"] },
+        clearScreen: false,
+      });
+    }
+  }
+  const outDir = join(rootDir, "dist", "server");
+  outputFileSync(join(outDir, ".env"), `APP_SECRET=${appSecret}\n`);
+  buildServer({
+    entryPoints: [join(serverDir, ".hathora", "store.ts")],
+    bundle: true,
+    platform: "node",
+    format: "esm",
+    outfile: join(outDir, "index.mjs"),
+    banner: {
+      js: "import { createRequire as topLevelCreateRequire } from 'module';\n const require = topLevelCreateRequire(import.meta.url);",
+    },
+  });
+}
+
 const rootDir = getProjectRoot(process.cwd());
 const clientDir = join(rootDir, "client");
 const serverDir = join(rootDir, "server");
@@ -129,28 +154,13 @@ if (command === "init") {
   install();
   startServer().then(startFrontends);
 } else if (command === "build") {
-  process.env.VITE_APP_ID = appId;
-  for (const dir of readdirSync(clientDir)) {
-    if (existsSync(join(clientDir, dir, "index.html"))) {
-      buildClient({
-        root: join(clientDir, dir),
-        build: { outDir: join(rootDir, "dist", "client", dir), target: ["esnext"] },
-        clearScreen: false,
-      });
-    }
+  if (!existsSync(join(serverDir, "impl.ts"))) {
+    console.error("Missing impl.ts, make sure to run hathora init first");
+  } else {
+    generate(rootDir, "templates/base");
   }
-  const outDir = join(rootDir, "dist", "server");
-  outputFileSync(join(outDir, ".env"), `APP_SECRET=${appSecret}\n`);
-  buildServer({
-    entryPoints: [join(serverDir, ".hathora", "store.ts")],
-    bundle: true,
-    platform: "node",
-    format: "esm",
-    outfile: join(outDir, "index.mjs"),
-    banner: {
-      js: "import { createRequire as topLevelCreateRequire } from 'module';\n const require = topLevelCreateRequire(import.meta.url);",
-    },
-  });
+  install();
+  build();
 } else {
   console.error(`Unknown command: ${command}`);
 }

